### PR TITLE
PYIC-7129: Remove old cimit lambda code

### PR DESCRIPTION
**Pull request by Wynndow**
### What changed

Remove old cimit lambda code

### Why did it change

We no longer call the cimit lambdas directly, instead using an API gateway. This means we can remove all the cruft.

### Issue tracking

- [PYIC-7129](https://govukverify.atlassian.net/browse/PYIC-7129)

[PYIC-7129]: https://govukverify.atlassian.net/browse/PYIC-7129?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/cimit-service/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java [hunk 1]
```diff
@@ -437,9 +284,13 @@ private HttpResponse<String> sendHttpRequest(HttpRequest cimitHttpRequest)
             LOGGER.info(LogHelper.buildLogMessage("Sending HTTP request to CiMit."));
             return httpClient.send(cimitHttpRequest, HttpResponse.BodyHandlers.ofString());
         } catch (InterruptedException e) {
+            LOGGER.error(
+                    LogHelper.buildErrorMessage(
+                            "Client interrupted sending http request to CiMit", e));
```
Comment (Joe-Edwards-GDS) [inline libs/cimit-service/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java]: I know we have it inconsistent throughout our code, but can we try to align on `CIMIT` as a stylisation going forward? (Likewise `SPOT` over `SPoT`)

I asked Chris C and he agrees - he thinks `CiMit` might have been a Slade-ism.

File: libs/cimit-service/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java [hunk 2]
```diff
@@ -128,34 +86,9 @@ public void submitVC(VerifiableCredential vc, String govukSigninJourneyId, Strin
...
-                if (lambdaExecutionFailed(response)) {
-                    logLambdaExecutionError(response, CI_STORAGE_PUT_LAMBDA_ARN);
-                    throw new CiPutException(FAILED_LAMBDA_MESSAGE);
-                }
-            }
+            var payload = OBJECT_MAPPER.writeValueAsString(new PostCiApiRequest(vc.getVcString()));
+
+            sendPostHttpRequest(POST_CI_ENDPOINT, govukSigninJourneyId, ipAddress, payload);
         } catch (JsonProcessingException e) {
             throw new CiPutException("Failed to serialize payload for post CI request.");
```
Comment (Joe-Edwards-GDS) [inline libs/cimit-service/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java]: One thing that Patrick spotted (https://gds.slack.com/archives/C05TJ52FZ8B/p1724411977038429) is that this error can also be thrown when deserializing the response and the error message is quite misleading.

The other thing from both these blocks is that we're losing the underlying exception, so we should make sure any relevant detail has been logged.

**Review by Joe-Edwards-GDS — COMMENTED** [review]
1000 lines deleted 😍

File: libs/cimit-service/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java (hunk 1 above)
Comment (Wynndow) [reply libs/cimit-service/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java]: ha - yep. Good to have an offical steer.

**Review by Joe-Edwards-GDS — APPROVED** [review]

**Pull request merged**